processor/src/main/java/org/mapstruct/ap/internal/model/ContainerMappingMethod.java

```diff
@@ -36,11 +36,8 @@
  *
  * @author Filip Hrisafov
  */
-public abstract class ContainerMappingMethod extends MappingMethod {
```

Comment:
This methods will remain:
* `getLoopVariableName`, `getIndex1Name` and `getIndex2Name` - for the arrays
* `getElementAssignment` - How the underlying element is assigned.
* `getResultElementType` - I think that this is the most important one. An abstract that needs to be implemented to get the `Type` of the result element. 
* The `selectionParameters`

I completely understand you not liking the long inheritance chain. Do you perhaps have some other suggestion? Maybe make the `MappingMethodWithFactory` (which will be renamed) generic and use that for the `MappingMethod` specifics?